cmd/podman/machine/start.go

```diff
@@ -79,7 +83,23 @@ func start(_ *cobra.Command, args []string) error {
 	if err := vm.Start(vmName, startOpts); err != nil {
 		return err
 	}
+	if err := changeDockerContext(); err != nil {
```

Comment:
If there is a agreement on this solution, there are few point which I think should be done:

* This should be only done for users who want this, maybe additional hidden flag ? Then `podman-desktop` can use this hidden flag by default `--change-docker-context` ? or something like that.
* After that should this be only done if `docker desktop` exists on host and has a version `>=4.13` ? Not for everybody since problem only persists for new users ?

Reply:
I don't understand the suggestion about moving the check/fix from start to installers. As Docker Desktop changes the context to $HOME/.docker/run/docker.sock each time it starts, it seems more convenient to me to have the context change command executed when the user starts the Podman machine.

Reply:
In all the versions I looked at this setting is only set during install / first launch. On all current versions, if you go with the defaults the docker cli will use /var/run/docker.sock (which symlinks to the user home sock),  If you ever installed 4.13.0 though, it would have written a different current context value to the ~/.docker/config.json, which sticks around until it's removed or changed. The other scenario which sets the context is if you pick Advanced settings and you disable /var/run/docker sock, at which point it will also write a context value to config.json.